lib/reply.js

```diff
@@ -211,7 +211,7 @@ Reply.prototype.type = function (type) {
 Reply.prototype.redirect = function (code, url) {
   if (typeof code === 'string') {
     url = code
-    code = this[kReplyHasStatusCode] ? this.res.statusCode : 302
+    code = 302
```

Comment:
What was not working about this conditional? Why would we ignore the `reply.code(301).redirect('/foo')` pattern? Wasn't `reply.redirect('/foo')` issuing a `302` before this change?

Reply:
Can you add an example where you were not getting a 302 with .redirect()? It seems you should have had.

Reply:
Nothing was wrong with it, but It's better and easier to use like this.

Right now I'm using the [nestjs](https://github.com/nestjs/nest) framework. Nest currently supports multiple routing libraries such as `express` and `fastify`. `nestjs` is setting the http status code default to `200` or `201` depending on the type of request (`GET`, `POST`). When I'm redirecting the request with `redirect` it returns it with a status of `200` instead of `302` this forms an issue because the browser won't redirect.

`express` is also overriding the status code to 302 by default and If you need to override it you provide it in the `res.redirect` like to: `res.redirect(404, '/not-found')`.

For more information read the issue: #1592

Reply:
You seem to be saying that Fastify is not defaulting the status code to `302` when using the `reply.redirect` method, but that is exactly what Fastify is doing.

Reply:
@jsumners Yes and no. If I did `res.code` in an earlier stage `reply.redirect` won't set the code to 302. It's only setting the code to 302 when It's not defined yet.

Reply:
Right.  Because you have already set a code for the response. If you set a code and then determined that you need a different code later in the cycle then you should set a new code.

Reply:
I know what you mean. But it feels like a design issue (although I completely understand the point of view you guys are having). But when I see `reply.redirect` I expect to have to browser redirected to the address I'm entering. But know that's not always the case because some other layer has already set a code. That's why it would be better to have the `redirect` set the 302 for you, except if you provide otherwise.

Reply:
> except if you provide otherwise.

You are describing that you _did_ provide otherwise. 

I think the answer is _either_ the current implementation, which as @delvedor pointed out, was created at community request, or the proposed implementation. I don’t think it’s possible to cover both cases. 

As I understand it, the issue is that a framework wrapping Fastify is doing `reply.code(200)` at some point but then subsequently issuing `reply.redirect('/foo')`. If that is the case, I vote to keep the current implementation and say the “bug” is in the wrapping framework, not Fastify.